otter/test/test_retry.py

```diff
@@ -456,13 +458,19 @@ class EffectfulRetryTests(SynchronousTestCase):
     # an object which calls a function with arguments specified as public
     # attributes, I guess...
 
+    def setUp(self):
+        self.dispatcher = ComposedDispatcher([
```

Comment:
+1. Yes please make custom dispatchers instead of using global one :) for the tests. Also, I wonder if there _should_ be a global dispatcher. But that is not in the scope of this PR.